components/console/commands.c

```diff
@@ -59,7 +63,8 @@ esp_err_t esp_console_init(const esp_console_config_t *config)
     if (s_config.hint_color == 0) {
         s_config.hint_color = ANSI_COLOR_DEFAULT;
     }
-    s_tmp_line_buf = calloc(config->max_cmdline_length, 1);
+    s_heap_alloc_caps = config->heap_alloc_caps;
```

Comment:
I think we should only set this if `config->heap_alloc_caps != 0` , some users might not be using `ESP_CONSOLE_CONFIG_DEFAULT`. So if users havent set  `heap_alloc_caps` we just keep using the default value `MALLOC_CAP_DEFAULT`

Reply:
good point

Reply:
updated